### PR TITLE
Gh12210 add parameter spreadsheet format

### PR DESCRIPTION
**Pull request by riadmen**

### CODE REVIEW

File: backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/impexp/spreadsheet/process/ExportToSpreadsheetProcess.java [hunk 1]
```diff
@@ -52,7 +73,7 @@ protected String doIt()
 		final File resultFile;
 
 		final SpreadsheetExportOptions spreadsheetExportOptions = getProcessInfo().getSpreadsheetExportOptions();
-		final SpreadsheetFormat spreadsheetFormat = spreadsheetExportOptions.getFormat();
+		//final SpreadsheetFormat spreadsheetFormat = spreadsheetExportOptions.getFormat();
```
Comment (metas-ts) [inline backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/impexp/spreadsheet/process/ExportToSpreadsheetProcess.java:76]:
```suggestion
```

File: backend/de.metas.business/src/main/java/de/metas/datev/interceptor/DatevAcctExport.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+		final List<I_DatevAcctExportLine> lines = queryBL.createQueryBuilder(I_DatevAcctExportLine.class)
+				.addInArrayFilter(I_DatevAcctExportLine.COLUMNNAME_DatevAcctExport_ID, datevAcctExport.getDatevAcctExport_ID())
+				.create()
+				.list();
+
+		for (final I_DatevAcctExportLine line : lines)
+		{
+			InterfaceWrapperHelper.delete(line);
+		}
```
Comment (metas-ts) [inline backend/de.metas.business/src/main/java/de/metas/datev/interceptor/DatevAcctExport.java:54]:
```suggestion
				.delete();
```

File: backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/impexp/spreadsheet/process/ExportToSpreadsheetProcess.java [hunk 3]
```diff
@@ -42,6 +43,26 @@
 public class ExportToSpreadsheetProcess extends JavaProcess
 {
 	final SpreadsheetExporterService spreadsheetExporterService = SpringContextHolder.instance.getBean(SpreadsheetExporterService.class);
+	private	SpreadsheetFormat spreadsheetFormat;
+
+	@Override
+	protected void prepare()
+	{
+		spreadsheetFormat = getProcessInfo().getSpreadsheetExportOptions().getFormat();
```
Comment (metas-ts) [inline backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/impexp/spreadsheet/process/ExportToSpreadsheetProcess.java:51]: why do we need this line?
i think you can loose this whole `prepare` method and use `@Param` instead. See e.g. `de.metas.impexp.process.C_DataImport_ImportAttachment#p_AD_AttachmentEntry_ID`

**Review by metas-ts — APPROVED** [review]
Please check my requests. Also, does the export actually work? asking because afais you didn't document that bit..

File: backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/impexp/spreadsheet/process/ExportToSpreadsheetProcess.java (hunk 1 above)
Comment (riadmen) [reply backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/impexp/spreadsheet/process/ExportToSpreadsheetProcess.java]: modification done

File: backend/de.metas.business/src/main/java/de/metas/datev/interceptor/DatevAcctExport.java (hunk 2 above)
Comment (riadmen) [reply backend/de.metas.business/src/main/java/de/metas/datev/interceptor/DatevAcctExport.java]: modification done

File: backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/impexp/spreadsheet/process/ExportToSpreadsheetProcess.java (hunk 3 above)
Comment (riadmen) [reply backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/impexp/spreadsheet/process/ExportToSpreadsheetProcess.java]: modification done

File: backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/impexp/spreadsheet/process/ExportToSpreadsheetProcess.java [hunk 4]
```diff
@@ -52,7 +58,13 @@ protected String doIt()
 		final File resultFile;
 
 		final SpreadsheetExportOptions spreadsheetExportOptions = getProcessInfo().getSpreadsheetExportOptions();
-		final SpreadsheetFormat spreadsheetFormat = spreadsheetExportOptions.getFormat();
+		SpreadsheetFormat spreadsheetFormat = getProcessInfo().getSpreadsheetExportOptions().getFormat();
+
+		if(p_SpreadsheetFormat != null)
+		{
+			spreadsheetFormat = SpreadsheetFormat.ofCode(p_SpreadsheetFormat);
+		}
```
Comment (metas-ts) [inline backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/impexp/spreadsheet/process/ExportToSpreadsheetProcess.java:66]:
```suggestion
		final SpreadsheetFormat spreadsheetFormat = CoalesceUtil.coalesceNotNull(
				SpreadsheetFormat.ofNullableCode(p_SpreadsheetFormat),
				spreadsheetExportOptions.getFormat());
```

**Review by metas-ts — CHANGES_REQUESTED** [review]
please check out the proposed refactoring

**Pull request merged**